reaper/main.go

```diff
@@ -1,12 +1,34 @@
 package main
 
 import (
+	"os"
+
 	"github.com/sirupsen/logrus"
 )
 
+const envLogLevel = "LOG_LEVEL"
+const defaultLogLevel = logrus.InfoLevel
+
 func main() {
+	logLevel := getLogLevel()
 	logrus.SetFormatter(&logrus.JSONFormatter{})
+	logrus.SetLevel(logLevel)
+
 	reaper := newReaper()
 	reaper.harvest()
 	logrus.Info("pod reaper is exiting")
 }
+
+func getLogLevel() logrus.Level {
+	levelString, exists := os.LookupEnv(envLogLevel)
+	if !exists {
+		return defaultLogLevel
+	}
+
+	level, err := logrus.ParseLevel(levelString)
+	if err != nil {
+		return defaultLogLevel
```

Comment:
This one is interesting!
How can I know that I've configured the logging level incorrectly? I feel like this should at least be logging the failure to configure the log level, but that feels almost cyclic to me. How do you feel about a `panic` here? It seems aggressive but it's make it very clear that something is not configured the way we want it.

This is another thing that I need to consider in the rule-refactor. Previously with rules being "loaded" when the reaper started, it failed very quickly if something was not configured correctly -- rather than when the rules are evaluated. I'll make a note of that on the other PR.

Reply:
I actually thought of this after I submitted the PR. 

You're 100% correct; we shouldn't hide errors. However, this is not a fatal error, and falling back to the default log level is reasonable. 

I am now logging the error before returning the default level.